site-builder/src/site/resource.rs

```diff
@@ -399,6 +384,43 @@ impl ResourceManager {
         )))
     }
 
+    /// Based on the ws-resources.yaml definition, this function derives the HTTP headers
+    /// based on wildcard matching, the headers that should be added to the HTTP response
+    /// of the resource defined by the resource_path.
```

Comment:
Suggestion: let's use the rustdoc convention for documentation here as well
```suggestion
    ///  Derives the HTTP headers for a resource based on the ws-resources.yaml.
    ///
    ///  Matches the path of the resource to the wildcard paths in the configuration to ....     
```